modules/permutiveRtdProvider.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * This module adds permutive provider to the real time data module
+ * The {@link module:modules/realTimeData} module is required
+ * The module will add custom targeting to ad units of specific partners
+ * @module modules/permutiveRtdProvider
+ * @requires module:modules/realTimeData
+ */
+import { getGlobal } from '../src/prebidGlobal.js'
+import { submodule } from '../src/hook.js'
+import { getStorageManager } from '../src/storageManager.js'
+import { deepSetValue, deepAccess, isFn, mergeDeep } from '../src/utils.js'
+import includes from 'core-js-pure/features/array/includes.js'
+
+export const storage = getStorageManager()
+
+function init (config, userConsent) {
+  return true
+}
+
+/**
+* Set targeting from cache and then try to wait for Permutive
+* to initialise to get realtime targeting
+*/
+export function initTargeting (reqBidsConfigObj, callback, customConfig) {
+  const permutiveOnPage = isPermutiveOnPage()
+  const config = mergeDeep({
+    waitForIt: false,
+    params: {
+      maxSegs: 500,
+      acBidders: [],
+      overwrites: {}
+    }
+  }, customConfig)
+
+  setTargeting(reqBidsConfigObj, config)
+
+  if (config.waitForIt && permutiveOnPage) {
+    window.permutive.ready(function () {
+      setTargeting(reqBidsConfigObj, config)
+      callback()
+    }, 'realtime')
+  } else {
+    callback()
+  }
+}
+
+function setTargeting (reqBidsConfigObj, config) {
```

Comment:
Please also consider conforming to the new segment syntax defined by the Prebid Taxonomy committee and the IAB. ee https://github.com/prebid/Prebid.js/issues/6057

I understand that right now this is a draft spec not implemented by the adapters, but... Chicken and Egg. If you start writing the segments now, the bidders can start picking them up when they're ready.

Reply:
Totally agree. We went for parity on how segments are passed into bidders as custom params today (today our clients add custom code to their sites, that sets bidder config as params). But we'd also prefer to use the proper FPD object instead. The issues we came across right now is that bidders do not accept and parse the new structure yet. 

In the example below data isn't pared as expected:
```javascript
pbjs.setConfig({
  fpd: {
    user: {
      data: [
        {
          "name": "www.dataprovider1.com",
          "ext": { "taxonomyname": "Example Proprietary Taxonomy ABC" },
          "segment": [
            { "id": "687" },
            { "id": "123" }
          ]
        },
        {
          "name": "www.dataprovider1.com",
          "ext": { "taxonomyname": "IAB Audience Taxonomy" },
          "segment": [
            { "id": "687" },
            { "id": "123" }
          ]
        }
      ]
    }
  }
})
```
results in:
![image](https://user-images.githubusercontent.com/5756475/107620193-3ce6b900-6c4c-11eb-83c0-36d483831114.png)

Based on you other comment, I assume the idea is that we'd use `fpd.user.ortb2.data` instead of `fpd.user.data`? So set segments like below:
```javascript
pbjs.setConfig({
  fpd: {
    user: {
      ortb2: {
        data: [
          {
            "name": "www.dataprovider1.com",
            "ext": { "taxonomyname": "Example Proprietary Taxonomy ABC" },
            "segment": [
              { "id": "687" },
              { "id": "123" }
            ]
          },
          {
            "name": "www.dataprovider1.com",
            "ext": { "taxonomyname": "IAB Audience Taxonomy" },
            "segment": [
              { "id": "687" },
              { "id": "123" }
            ]
          }
        ]
      }
    }
  }
})
```

What we could do is keep the current structure to make sure targeting works in the here and now, but then also set `fpd.user.ortb2.data` in parallel for when adapters accept that structure. Is this what you recommend? There are still a few open questions on our end to decide how to expose the right `name` and `taxonomyname` and what these values should be for different use-cases. If you're happy with it, I'd suggest to go ahead with v1 which takes the same approach as the current setup and we'll then follow up with a PR for v2 which adds support for `fpd.user.ortb2.data` - ready for when individual adapters start picking it up.

Reply:
I agree that the adapters aren't as of yet accepting user.data.segment. But I know that conversation is happening right now in the Magnite(rubicon) exchange team, so it won't be long before we can update the rubicon adapter to pass in the segments where they want them.

So for now, I'm suggesting pass the data BOTH ways: as bidder.params.visitor and as ortb2.user.data. Bidders will ignore the latter for now, but it being there from the start kickstarts the chicken/egg problem.

For the record, 'fpd' setConfig will be gone once https://github.com/prebid/Prebid.js/pull/6293 is merged. So I recommend ignoring 'fpd' starting now -- I recommend doing a "getConfig" on ortb2, adding your stuff to the user.data array, and setConfiging it back.

```
pbjs.setConfig({
    ortb2: {
      ... stuff that publisher may have already added ...
      user: {
        data: [ { ... } ]
     }
  }
```

Reply:
That makes perfect sense, thanks @bretg. Is it ok if we follow up with a separate PR for this? Implementing the change should be quick on our end. But we'll need to first understand what the appropriate values for `name` and `taxonomyname` are for the different use-cases we'll support with this module. Great to hear that the Magnite team is already working on supporting `user.data.segment`!

Also, thanks for your note on `getConfig` before calling `setConfiging` - we want to make sure we're not overwriting any other segment data objects